### PR TITLE
refactor: use pytest fixtures to reduce repetition in tests

### PR DESCRIPTION
**Pull request by Aswinr24**
# Description

Refactored the test cases for plotting voltage components to use pytest fixtures, reducing code duplication and improving test maintainability.

Related to #4502 

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [x] Code refactor (test improvements to reduce repetition)

# Key checklist:

- [x] No style issues: `$ pre-commit run` (or `$ nox -s pre-commit`) (see [CONTRIBUTING.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CONTRIBUTING.md#installing-and-using-pre-commit) for how to set this up to run automatically when committing locally, in just two lines of code)
- [x] All tests pass: `$ python run-tests.py --all` (or `$ nox -s tests`)
- [x] The documentation builds: `$ python run-tests.py --doctest` (or `$ nox -s doctests`)

You can run integration tests, unit tests, and doctests together at once, using `$ nox -s quick`.

## Further checks:

- [ ] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works

### CODE REVIEW

File: tests/unit/test_plotting/test_plot_voltage_components.py [hunk 1]
```diff
@@ -7,77 +11,37 @@
...
-            t, V = ax.get_lines()[0].get_data()
-            np.testing.assert_array_equal(t, sol["Time [h]"].data)
-            np.testing.assert_array_equal(V, sol["Battery voltage [V]"].data)
 
-        _, ax = plt.subplots()
-        _, ax_out = sol.plot_voltage_components(ax=ax, show_legend=True)
-        assert ax_out == ax
+@pytest.mark.parametrize(
+    "from_solution", [True, False], ids=["from_solution", "from_simulation"]
+)
```
Comment (arjxn-py) [inline tests/unit/test_plotting/test_plot_voltage_components.py:24]: pre-commit is not happy when you want to keep it in a single line?

**Review by arjxn-py — COMMENTED** [review]
Nicely Done :)

File: tests/unit/test_plotting/test_plot_voltage_components.py [hunk 2]
```diff
@@ -7,77 +11,37 @@
...
+@pytest.mark.parametrize("split_by_electrode", [True, False], ids=["split", "no_split"])
```
Comment (prady0t) [inline tests/unit/test_plotting/test_plot_voltage_components.py:25]:
```suggestion
@pytest.mark.parametrize(
    "split_by_electrode",
    [True, False],
    ids=["with_split_by_electrode", "with_no_split_by_electrode"],
)
@pytest.mark.parametrize(
    "from_solution", [True, False], ids=["from_solution", "from_simulation"]
)
```
Before : 

```
[gw6] [ 20%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_without_solution 
[gw2] [ 40%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[no_split-from_solution] 
[gw3] [ 60%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[no_split-from_simulation] 
[gw1] [ 80%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[split-from_simulation] 
[gw0] [100%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[split-from_solution] 
```
After : 

```
[gw7] [ 20%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_without_solution 
[gw4] [ 40%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[from_solution-with_no_split_by_electrode] 
[gw0] [ 60%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[from_solution-with_split_by_electrode] 
[gw3] [ 80%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[from_simulation-with_no_split_by_electrode] 
[gw1] [100%] PASSED tests/unit/test_plotting/test_plot_voltage_components.py::test_plot_voltage_components[from_simulation-with_split_by_electrode] 
```
Test names are a bit better now.

File: tests/unit/test_plotting/test_plot_voltage_components.py (hunk 1 above)
Comment (prady0t) [reply tests/unit/test_plotting/test_plot_voltage_components.py:24]: No it changes it to multi-line.

**Comment by prady0t** [comment]
Nice work @Aswinr24! Tests look much cleaner now.

File: tests/unit/test_plotting/test_plot_voltage_components.py (hunk 1 above)
Comment (Aswinr24) [reply tests/unit/test_plotting/test_plot_voltage_components.py:29]: Yes got changed to multi line in pre-commit

**Comment by Aswinr24** [comment]
Hey thanks for the review @prady0t @arjxn-py !
Should I add any line to the [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document this change?

**Comment by kratman** [comment]
> Should I add any line to the [CHANGELOG.md](https://github.com/pybamm-team/PyBaMM/blob/develop/CHANGELOG.md) to document this change?

Not needed. Users don't need to know we updated tests

**Review by arjxn-py — APPROVED** [review]

File: tests/unit/test_plotting/test_plot_voltage_components.py [hunk 3]
```diff
@@ -1,3 +1,7 @@
+#
+# Tests for the voltage plot components functions
+#
+
```
Comment (kratman) [inline tests/unit/test_plotting/test_plot_voltage_components.py:4]: We don't need comments like this, the file name already says this

Comment (kratman) [reply tests/unit/test_plotting/test_plot_voltage_components.py:4]:
```suggestion
```

**Review by kratman — APPROVED** [review]

**Pull request merged**